### PR TITLE
[WIP] Teacher model

### PR DESCRIPTION
**Pull request by arnoFleming**

### CODE REVIEW

**Comment by Awadje** [comment]
- Tests are now green: Removed whitespace in Teacher Factory 
- Generated a migration file to add a name column to the Teacher table

**Comment by arnoFleming** [comment]
I'd like to see (at least presence) validations for the name

**Comment by Awadje** [comment]
- Generate a migration file to add a user reference column to the Teacher table to pass teacher spec
- Add **belongs to** relation to User in Teacher to pass teacher spec
- Update teachers Factory to be able to build successfully to **pass teacher spec**
       _- Create users Factory to be able to be able to successfully build teachers Factory_
- Add validation for name in Teacher
- Add validation test to teacher spec

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,2 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
-  devise :database_authenticatable, :registerable,
```
Comment (arnoFleming) [inline app/models/user.rb]: Why is this removed?

File: db/schema.rb [hunk 2]
```diff
@@ -10,7 +10,9 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20170508092641) do
+
```
Comment (arnoFleming) [inline db/schema.rb]: what's with the extra newline/whitespace - Did Rails add this?

File: spec/models/teacher_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+RSpec.describe Teacher, type: :model do
+  # let(:user1) { create :user }
+  # let(:teacher1) { create :teacher, user: user1 }
+  # it "belongs to a user" do
+  #     expect(teacher1.user_id).to eq(user1.id)
+  #   end
+
+  describe "validations" do
+    it "is invalid without a name" do
```
Comment (arnoFleming) [inline spec/models/teacher_spec.rb]: can you rewrite this example the shoulda way? (see https://github.com/thoughtbot/shoulda-matchers)

File: app/models/user.rb (hunk 1 above)
Comment (Awadje) [inline app/models/user.rb]: It gave errors so I commented it out and removed them at one point for testing purposes, forgot to put it back.

File: db/schema.rb (hunk 2 above)
Comment (Awadje) [inline db/schema.rb]: yes...

File: spec/models/teacher_spec.rb (hunk 3 above)
Comment (Awadje) [inline spec/models/teacher_spec.rb]: Done!

File: app/models/teacher.rb [hunk 4]
```diff
@@ -1,2 +1,5 @@
 class Teacher < ApplicationRecord
+  belongs_to :user
+  attr_accessor :name
+  validates_presence_of :name, on: :create
```
Comment (arnoFleming) [inline app/models/teacher.rb]: Also on edit I'd like the name to be present. Please remove the `on: ` part

File: spec/models/teacher_spec.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe Teacher, type: :model do
+  let(:user1) { create :user }
+  let(:teacher1) { create :teacher, user: user1 }
+
+  it "belongs to a user" do
+    expect(teacher1.user_id).to eq(user1.id)
```
Comment (arnoFleming) [inline spec/models/teacher_spec.rb]: This test is the same as the test on line 17 I think.

Comment (arnoFleming) [inline spec/models/teacher_spec.rb]: You do know that @Tenicia  is working on this ticket, right?

@Tenicia can you review this PR?
